src/types/instantsearch.ts

```diff
@@ -4,11 +4,22 @@ import {
   SearchParameters,
   PlainSearchParameters,
 } from 'algoliasearch-helper';
-import { Index } from '../widgets/index/index';
 import { InsightsClient as AlgoliaInsightsClient } from './insights';
 import { Widget, UiState } from './widget';
-
-export type InstantSearchOptions = any;
+export { default as InstantSearch } from '../lib/InstantSearch';
+
+export type SearchClient = Pick<Client, 'search' | 'searchForFacetValues'>;
+
+export type InstantSearchOptions = {
```

Comment:
We can copy the descriptions and default values (with `@default`) from the [`instantsearch` doc](https://www.algolia.com/doc/api-reference/widgets/instantsearch/js/#options). That's helpful when working on the codebase and to onboard people.

Reply:
This option can be inlined inside the module now that it's a TypeScript one.